packages/plans-grid-next/src/hooks/data-store/use-grid-plan-for-spotlight.ts

```diff
@@ -5,27 +5,27 @@ import { GridPlan, PlansIntent } from '../../types';
 const SPOTLIGHT_ENABLED_INTENTS = [ 'plans-default-wpcom' ];
 
 interface Params {
-	gridPlansForFeaturesGrid: GridPlan[];
+	gridPlans: GridPlan[] | null;
 	intent?: PlansIntent;
 	isSpotlightOnCurrentPlan?: boolean;
 	sitePlanSlug?: string | null;
 }
 
 const useGridPlanForSpotlight = ( {
-	gridPlansForFeaturesGrid,
+	gridPlans,
 	intent,
 	isSpotlightOnCurrentPlan,
 	sitePlanSlug,
```

Comment:
Aside: I think we can effectively get rid of `sitePlanSlug` from everywhere now, right? (since `useCurrentPlan( { siteId } )` would retrieve the same 🤔

Reply:
Updated: [405424c](https://github.com/Automattic/wp-calypso/pull/87478/commits/405424c74456a496c8de5d9335101709200c1048)

On a related note, just looking at the implementation for `useCurrentPlan`:

https://github.com/Automattic/wp-calypso/blob/9aad849346d88885cbccbad8dcc0226613202bae/packages/data-stores/src/plans/hooks/use-current-plan.ts#L5-L19

At the moment it can return `undefined`. Doesn't every site need at least one plan, and therefore this hook should never return `undefined`? 🤔 

I'm also unsure about `siteId` being `undefined` or `null`. Can we narrow that type to always expect a value?

Reply:
> Along the same lines, should we make `siteId` a required value?

The short answer, we can't. 

I think it's a trap to start thinking of these hooks (and components) as things that live completely separate from the ecosystem (wpcom). We decouple and extract things into packages, but they all need to come together to serve this bigger frontend. In this ecosystem, a "selected-site" is a real and unique concept, and it can be `undefined` e.g. when a user goes through the signup flow or until a site hasn't been created.

I think this goes back to our earlier chat about refactoring `siteId` to `selectedSiteId`. The type that's propagated throughout is `number | null | undefined`, which comes from top-level selectors, like `getSelectedSiteId`. It cannot possibly be undefined if a "selected-site" exists i.e. a site without an ID is not a valid thing. Hence I find the terms dependent and not easily separable - meaning, it makes more sense for something to accept a "selected-site ID" that's undefined than it does to accept a "site ID" that's undefined.

That's my thinking. Curious, does this make you reconsider at all refactoring `selectedSiteId` back to `siteId` earlier? 🤔

Reply:
As for `useCurrentPlan` above accepting `siteId?: string | number | null` - it's because the actual variable for the API path ought to be `siteIdOrSlug` (it can be either a site ID or slug). I haven't had a chance to update all of these hooks, but for some I did e.g. in `/data-stores/site` (same root endpoint).

Reply:
> In this ecosystem, a "selected-site" is a real and unique concept, and it can be undefined e.g. when a user goes through the signup flow or until a site has been created

Thanks for highlighting this example.

Wdyt about defining a new type `SiteIdOrSlug?: string | number | null` that includes a comment describing the `undefined` scenario eg. `"Can be undefined in certain contexts, for example during signup"`

> it makes more sense for something to accept a "selected-site ID" that's undefined than it does to accept a "site ID" that's undefined

Yep I do see what you mean. Do you think the new type + comment for a `siteId` prop might avoid some of that confusion?

> does this make you reconsider at all refactoring selectedSiteId back to siteId earlier?

Hmm. So I think the scenario I come back to that helps clarify the naming for me is the hypothetical 'multi-site editor' which displays sites side by side:

```
Site = ({ siteId }) => {
  // single site context
}

MultiSiteEditor = () => {
  // multi-site context with a selected site

  return (
    <>
      <Site siteId={selectedSiteId} />
      <Site siteId={selectedSiteId} />
      <Site siteId={selectedSiteId} />
      ...
```

In this UI users could select a site to delete (for example), and this `selectedSiteId` would exist in the context of `MultiSiteEditor`, but by having a prop `selectedSiteId` for `Site`, we're introducing context within `Site` that it shouldn't know about -- an instance of `Site` can exist without anything being 'selected'.

Wdyt 🤔